src/graphql/status.rs

```diff
@@ -198,29 +179,22 @@ impl GigantoStatusQuery {
 #[Object]
 impl GigantoConfigMutation {
     #[allow(clippy::unused_async)]
-    async fn set_giganto_config<'ctx>(
-        &self,
-        ctx: &Context<'ctx>,
-        field: UserConfig,
-    ) -> Result<String> {
+    async fn set_giganto_config<'ctx>(&self, ctx: &Context<'ctx>, draft: String) -> Result<String> {
         let cfg_path = ctx.data::<String>()?;
+
+        let config_toml = fs::read_to_string(cfg_path).context("toml not found")?;
+        let config: GigantoConfig = toml::from_str(&config_toml)?;
+
+        let config_draft: GigantoConfig = toml::from_str(&draft)?;
```

Comment:
I suggest to put this line at the top of this function, because this in the course of parsing the request parameter, the parameter is validated, and parameter validations are usually completed before the server starts its business logic.